crates/nu-parser/src/lex.rs

```diff
@@ -438,6 +438,19 @@ fn lex_internal(
             // comment. The comment continues until the next newline.
             let mut start = curr_offset;
 
+            // if previous token is eol, we just delete that token
+            // so at high level, it makes the following transform:
+            //
+            // ls             ls  # | le
+            // # | le    ==>  | length
+            // | length
+            //
+            // And it makes lite_parsing easiler to generate a correct LiteBlock
```

Comment:
maybe we can just link to this PR instead of adding a huge comment? :thinking:

Reply:
Normally I'm not a fans of link a pr as a comment, it's poor explained if the it's not too complex